libs/ts-rest/core/src/lib/client.ts

```diff
@@ -250,6 +254,8 @@ export const fetchApi = ({
     },
     body:
       body !== null && body !== undefined ? JSON.stringify(body) : undefined,
+    rawBody: body,
+    contentType: route.method !== 'GET'  ? route.contentType : undefined,
```

Comment:
In mutations, `route.contentType` can be undefined, and when it is so, it is assumed to be `application/json`, but in this case we are still passing `undefined` here.

Since at the moment `contentType` can be `'application/json' | 'multipart/form-data' | undefined`, and the form-data case is already handled in the above code, we can hard code this to `application/json` just like on line 252 (we are also forcing the body to be `JSON.stringify`-ied so it can't be anything other than `application/json` at the moment)

Reply:
I think it might also make sense to change that line 252 thing to `...(route.method !== 'GET' && { 'content-type': 'application/json' })`, it doesn't really make sense to send a 'content-type' header with a GET request.

Reply:
Ah, so maybe make the `contentType` an optional property then?

Reply:
@michaelangrivera seems like it's already optional since it's inheriting from `AppRouteMutation['contentType']` which is also optional. But I think we should only pass `undefined` if method is GET, otherwise it must always be set. The core library is allowing undefined in the contract definition, but it assumes that as `application/json`, so we should also expose it as such to the custom client.

Reply:
Makes sense! I will update!

Reply:
Actually, `contentType` is only inheriting the _types_ from `AppRouteMutation['contentType']`, not the optional property key itself, so `contentType` is _not_ marked as optional, so we are OK in this sense (using the types from `AppRouteMutation['contentType']` for now). 

I've changed the value to always be `application/json` or `multipart/form-data`, though. Please take a look!

Reply:
Looks good!

And yeah, my bad on using "optional" and having a type of a undefined interchangeably. I meant that the key in `AppRouteMutation` being optional makes the type of `AppRouteMutation['contentType']` include undefined in its union type. So yes, the new param `contentType` is not optional (rightly so), but we can still pass undefined to it in the case of GET requests.